firedrake/ensemble.py

```diff
@@ -21,37 +23,28 @@ def __init__(self, comm, M):
 
         rank = comm.rank
 
-        # User global comm
+        # Global comm
         self.global_comm = comm
         # Internal global comm
-        self._global_comm = internal_comm(comm, self)
+        self._comm = internal_comm(comm, self)
```

Comment:
I'm not sure I like the inconsistency between `self.global_comm` and `self._comm` and between `self.comm` and `self._spatial_comm` but I guess that can't be helped seeing as:
- the garbage collection assumes that an object is defined over `obj._comm` and
- a user expects `obj.comm` to be the "spatial" comm associated to the object.

Reply:
I understand your point, but `_comm` really is an internal implementation detail. I will leave this unresolved for wider discussion, just in case.